src/matrixUtils.js

```diff
@@ -104,8 +104,8 @@ async function discoverHomeserverUrl(serverName) {
     if (delegatedHostname) {
         const parsed = parseHostnameAndPort(delegatedHostname);
 
-        // Don't continue if we consider the hostname part to resolve to our blacklisted IP ranges
-        if (await utils.isDomainBlacklisted(parsed.hostname)) {
+         // Don't continue if we consider the hostname part to resolve to our blacklisted IP ranges
+        if (utils.isBlacklisted(utils.resolveDomain(parsed.hostname))) {
```

Comment:
There is an `await` missing in front of the `utils.resolveDomain()` call so it passes a `Promise` to `utils.isBlacklisted()` where it will be evaluated to an empty array. The same is true for line 58.

Reply:
Fixed this. Thanks!